spyderlib/widgets/arrayeditor.py

```diff
@@ -414,11 +414,14 @@ def _sel_to_text(self, cell_range):
         """Copy an array portion to a unicode string"""
         row_min, row_max, col_min, col_max = get_idx_rect(cell_range)
         _data = self.model().get_data()
-        output = io.StringIO()
+        if PY3:
+            output = io.BytesIO()
+        else:
+            output = io.StringIO()
         np.savetxt(output,
                   _data[row_min:row_max+1, col_min:col_max+1],
                   delimiter='\t')
-        contents = output.getvalue()
+        contents = output.getvalue().decode('utf-8')

```

Comment:
The decode part will be executed for both python 2 and 3. Was the functionnality bugged in python2 too, then ?